src/PathOrderOptimizer.cpp

```diff
@@ -0,0 +1,61 @@
+//Copyright (c) 2020 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#include "PathOrderOptimizer.h" //The definitions we're implementing here.
+#include "sliceDataStorage.h" //For SliceLayerPart.
+
+//Since the PathOrderOptimizer is a template class, we will only implement the template specializations in this file.
+
+namespace cura
+{
+
+template<>
+ConstPolygonRef PathOrderOptimizer<ConstPolygonRef>::getVertexData(ConstPolygonRef path)
+{
+    return path;
+}
+
+template<>
+ConstPolygonRef PathOrderOptimizer<PolygonRef>::getVertexData(PolygonRef path)
+{
+    return path;
+}
+
+template<>
+ConstPolygonRef PathOrderOptimizer<const SkinPart*>::getVertexData(const SkinPart* path)
+{
+    return path->outline.outerPolygon();
+}
+
+template<>
+ConstPolygonRef PathOrderOptimizer<const SliceLayerPart*>::getVertexData(const SliceLayerPart* path)
+{
+    if(!path->insets.empty())
+    {
+        return path->insets[0][0];
+    }
+    else
+    {
+        return path->outline.outerPolygon();
+    }
+}
+
+template<>
+ConstPolygonRef PathOrderOptimizer<const SupportInfillPart*>::getVertexData(const SupportInfillPart* path)
+{
+    return path->outline.outerPolygon();
+}
+
+template<>
+ConstPolygonRef PathOrderOptimizer<const std::vector<ExtrusionJunction>*>::getVertexData(const std::vector<ExtrusionJunction>* path)
+{
+	cached_vertices.emplace_back();
```

Comment:
2 extra spaces, alignment off.

Reply:
Fixed. It was actually a tab instead of a space.

My IDE was doing that sometimes because my default is configured to indent with tabs for personal projects. I had made an exception for CuraEngine, but only for C++. Sometimes it would randomly detect a file to be C instead of C++ and then it would indent. I've now configured the IDE with another exception for C and header files. Why it would ever think that this template-heavy code is not C++ is a mystery to me, but it shouldn't happen again.